src/NuGet.Core/NuGet.Configuration/Settings/Items/CertificateItem.cs

```diff
@@ -0,0 +1,115 @@
+// Copyright(c) .NET Foundation.All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Xml.Linq;
+using NuGet.Common;
+using NuGet.Shared;
+
+namespace NuGet.Configuration
+{
+    public sealed class CertificateItem : SettingItem, IEquatable<CertificateItem>
+    {
+        public string Fingerprint => Attributes[ConfigurationConstants.Fingerprint];
+
+        public HashAlgorithmName HashAlgorithm => CryptoHashUtility.GetHashAlgorithmName(Attributes[ConfigurationConstants.HashAlgorithm]);
+
+        public bool AllowUntrustedRoot
+        {
+            get
+            {
+                if (bool.TryParse(Attributes[ConfigurationConstants.AllowUntrustedRoot], out var parsedAttribute))
+                {
+                    return parsedAttribute;
+                }
+
+                return false;
+            }
+        }
+
+        protected override HashSet<string> RequiredAttributes => new HashSet<string>() { ConfigurationConstants.Fingerprint, ConfigurationConstants.HashAlgorithm, ConfigurationConstants.AllowUntrustedRoot };
+
+        public CertificateItem(string fingerprint, HashAlgorithmName hashAlgorithm, bool allowUntrustedRoot = false)
+            : base()
+        {
+            if (string.IsNullOrEmpty(fingerprint))
+            {
+                throw new ArgumentNullException(nameof(fingerprint));
+            }
+
+            if (hashAlgorithm == HashAlgorithmName.Unknown)
+            {
+                throw new ArgumentException(Resources.UnknownHashAlgorithmNotSupported);
+            }
+
+            AddAttribute(ConfigurationConstants.Fingerprint, fingerprint);
+            AddAttribute(ConfigurationConstants.HashAlgorithm, hashAlgorithm.ToString());
+            AddAttribute(ConfigurationConstants.AllowUntrustedRoot, $"{allowUntrustedRoot}");
+        }
+
+        internal CertificateItem(XElement element, SettingsFile origin)
+            : base(element, origin)
+        {
+            if (HashAlgorithm == HashAlgorithmName.Unknown)
+            {
+                throw new NuGetConfigurationException(string.Format(CultureInfo.CurrentCulture, Resources.UserSettings_UnableToParseConfigFile,
+                    string.Format(CultureInfo.CurrentCulture, Resources.UnsupportedHashAlgorithm, Attributes[ConfigurationConstants.HashAlgorithm]),
+                    origin.ConfigFilePath));
+            }
+        }
+
+        internal override SettingBase Clone()
+        {
+            var newItem = new CertificateItem(Fingerprint, HashAlgorithm, AllowUntrustedRoot);
+
+            if (Origin != null)
+            {
+                newItem.SetOrigin(Origin);
+            }
+
+            return newItem;
+        }
+
+        public bool Equals(CertificateItem other)
```

Comment:
May be I'm missing something... but why there are different impl for Equals and DeepEquals? I didnt even know there is some terminology as DeepEquals :)

Reply:
For any `SettingItem`: two items that have `Equals` true means that both of those items refer to the same setting, this does not mean that the items have the exact same values (this is what `DeepEquals` is for).

Think for example of an `AddItem`... `<add key='key1' value='val' /> and `<add key='key1' value='differentVal' /> would have `Equals() == true` but `DeepEquals() == false`.